chess/__init__.py

```diff
@@ -2916,6 +2916,29 @@ def is_zeroing(self, move: Move) -> bool:
         touched = BB_SQUARES[move.from_square] ^ BB_SQUARES[move.to_square]
         return bool(touched & self.pawns or touched & self.occupied_co[not self.turn] or move.drop == PAWN)
 
+    def is_promoting(self, move: Move) -> bool:
+        """
+        Checks whether the *move* is either a white pawn or a black pawn
+        attempting to promote.
+
+        Returns ``True`` only if all conditions are met. These conditions
+        include that the moving piece is actually a pawn that is attempting to
+        jump from the pre-last to the last rank and is not pinned to its king.
+        Also, the target square must be vacant or a legal capture.
+        """
+        if piece != PAWN:
```

Comment:
we need the following line at the beginning of the function, right?
`piece = self.piece_type_at(move.from_square)`
otherwise, piece at line 2929 is not defined.